### ISSUE TITLE
Multi GPU Problem

### ISSUE THREAD

**Issue by lhbrichard** [casperhansen/fake-news-reasoning#1]
i try to use multi-gpu strategy by using nn.DataParallel function. And i found the shape of logits may be [number_of_gpu * batch_size, -1], but the shape of labels is [batch_size, -1], so the loss can't be calculated properly. Did u meet the same problem or have u ever tried multi-gpu strategy to train the model?
![屏幕快照 2021-10-12 下午2 49 39](https://user-images.githubusercontent.com/33123730/136906241-60cbe8e3-0d63-4356-81f5-ccb69623851e.png)

**Comment by casperhansen** [comment]
We unfortunately have not tried to use a multi gpu strategy for this work.

**Comment by lhbrichard** [comment]
> We unfortunately have not tried to use a multi gpu strategy for this work.

thanks for answering. there is another question doubted me, is that, why u choose f1 macro on validation set to decide which step is the best result, instead of validation loss.

**Comment by casperhansen** [comment]
Thanks for you question. We could very well have chosen to stop based on the validation loss too, but chose the f1 macro as done by the MultiFC paper (from which we get the datasets).
